src/SPC/builder/Extension.php

```diff
@@ -486,18 +486,51 @@ public function isBuildStatic(): bool
         return $this->build_static;
     }
 
+    /**
+     * Get the library dependencies that current extension depends on.
+     *
+     * @param bool $recursive Whether it includes dependencies recursively
+     */
+    public function getLibraryDependencies(bool $recursive = false): array
+    {
+        $ret = array_filter($this->dependencies, fn ($x) => $x instanceof LibraryBase);
+        if (!$recursive) {
+            return $ret;
+        }
+
+        $deps = [];
+
+        $added = 1;
+        while ($added !== 0) {
+            $added = 0;
+            foreach ($ret as $depName => $dep) {
+                foreach ($dep->getDependencies(true) as $depdepName => $depdep) {
+                    if (!array_key_exists($depdepName, $deps)) {
+                        $deps[$depdepName] = $depdep;
+                        ++$added;
+                    }
+                }
+                if (!array_key_exists($depName, $deps)) {
+                    $deps[$depName] = $dep;
+                }
+            }
+        }
+
+        if (array_key_exists(0, $deps)) {
+            $zero = [0 => $deps[0]];
```

Comment:
what is this for?

Reply:
Oops, it's a previous buggy code workaround